client/js/common/application_view.js

```diff
@@ -127,6 +127,11 @@ appl.def('readable_date', function(date) {
   return momentTz(date).format("MM/DD/YY")
 })
 
+appl.def('payments_readable_date', function(date) {
+	if(!date) return
+	return moment.tz(date, "YYYY-MM-DD HH:mm:ss", ENV.nonprofitTimezone || 'UTC').format("MM/DD/YY")
+})
```

Comment:
I created this specific function to handle the display for the payments. When I worked on the conversion coming from the server, I did the changes only on the queries for the supporters and for the payments lists. The dates from the supporters list are not affected by the same problem because they don't get converted on the UI in the same way as the dates from the payments list.

Just for us to keep track of what was going on:

Let's say there was a payment that happened at **06/24/21 0:00am PDT**.

It gets recorded as **06/24/21 7:00am UTC**.

From the backend, we get already **06/24/21 0:00am**, because it is already converted on the query.

On the UI, we are receiving a **06/23/21 21:00pm**, which was being converted from UTC to my time zone.

After that, the first function `.tz(date, "YYYY-MM-DD HH:mm:ss", 'UTC')` was making it become **06/24/21 0:00am UTC**. The last function applied to this result, `.tz(ENV.nonprofitTimezone || 'UTC')`, was making it become  **06/23/21 17:00pm PDT**.

The current changes make the given time become **06/24/21 0:00am PDT**, which is the correct time.